tests/unit/adapters/osf-adapter-test.js

```diff
@@ -1,12 +1,401 @@
+            adapterOptions: {
+                url: url
+            }
+        }),
+        'createRecord'
+    );
+    assert.equal(url, result);
+});
+
+test('#_buildRelationshipURL uses relationshipLinks', function(assert) {
+    let url = 'https://api.osf.io/v2/users/me/nodes/';
+    let adapter = this.subject();
+    let user = FactoryGuy.make('user', {
+        links: {
+            relationships: {
+                nodes: {
+                    links: {
+                        related: {
+                            href: url
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    let result = adapter._buildRelationshipURL(
+        user._internalModel.createSnapshot({
+            adapterOptions: {
+                url: url
+            }
+        }),
+        'nodes'
+    );
+    assert.equal(url, result);
+});
+
+test('#_createRelated maps over each createdSnapshots and adds records to the parent\'s canonical state', function(assert) {
+    this.inject.service('store');
+    let store = this.store;
+
+    let node = FactoryGuy.make('node');
+    Ember.run.begin();
+    let contributors = [
+        store.createRecord('contributor', {
+            title: 'Foo'
+        }),
+        store.createRecord('contributor', {
+            title: 'Bar'
+        })
+    ];
+    Ember.run.end();
+    node.get('contributors').pushObjects(contributors);
+    let saveStubs = contributors.map(c => this.stub(c, 'save', () => {
+        return new Ember.RSVP.Promise((resolve) => resolve());
+    }));
+
+    var addCanonicalStub = this.stub();
+    this.stub(node, 'resolveRelationship', () => {
+        return {
+            addCanonicalRecord: addCanonicalStub
+        };
+    });
+
+    Ember.run(() => {
+        node.save().then(() => {
+            saveStubs.forEach(s => assert.ok(s.called));
+            assert.ok(addCanonicalStub.calledTwice);
+            // Can't use calledWith because sinon's deepEqual creates
+            // infinite recursive calls when comparing the Ember DS.Models
+            assert.equal(addCanonicalStub.args[0][0], contributors[0]);
+            assert.equal(addCanonicalStub.args[1][0], contributors[1]);
+        }, () => {
+            // Fail
+            assert.ok(false);
+        });
+    });
+});
+
+test('#_addRelated defers to _doRelatedRequest and adds records to the parent\'s canonical state', function(assert) {
+    let node = FactoryGuy.make('node');
+    let institution = FactoryGuy.make('institution');
+    node.get('affiliatedInstitutions').pushObject(institution);
+
+    var doRelatedStub = this.stub(OsfAdapter.prototype, '_doRelatedRequest', () => {
+        return new Ember.RSVP.Promise(resolve => resolve());
+    });
+    var relation = node.resolveRelationship('affiliatedInstitutions');
+    relation.hasLoaded = true;
+    var addCanonicalStub = this.stub(relation, 'addCanonicalRecord');
+
+    Ember.run(() => {
+        node.save().then(() => {
```

Comment:
Normally `node.save` does ajax work. Where *does* that request go during a unit test scenario?